### PR TITLE
chore(path): release `path@1.0.0-rc.1`

### PR DESCRIPTION
**Pull request by iuioiua**
Toward #4922

### CODE REVIEW

File: deno.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "@std/media-types": "jsr:@std/media-types@^1.0.0-rc.1",
     "@std/msgpack": "jsr:@std/msgpack@^0.224.2",
     "@std/net": "jsr:@std/net@^0.224.2",
-    "@std/path": "jsr:@std/path@^0.225.2",
+    "@std/path": "jsr:@std/path@1.0.0-rc.1",
```
Comment (kt3k) [inline deno.json:42]: 👍

**Review by kt3k — APPROVED** [review]
LGTM 👍

**Comment by kt3k** [comment]
Let's land this first and keep discussing #4954 during RC (Breaking changes during RC versions are common practice to my knowledge).

**Comment by iuioiua** [comment]
Frankly, breaking changes during the release candidate phase seems to conflict with the objective of a release candidate, and I disagree with it. Nevertheless, I'm merging this PR so we can progress with stabilization.

**Pull request merged**